### PR TITLE
[WIP] Add support for new SBP ephemeris messages

### PR DESCRIPTION
**Pull request by Paakkit**
This PR adds support for new SBP ephemeris messages for sb2rinex.

- GPS and GLO [done]
- BDS and GAL [done]

### CODE REVIEW

**Comment by Paakkit** [comment]
@mbavaro I started looking into the sb2rinex conversion.
Hopefully I'm on the right track with this PR.

Beidou and Galileo shouldn't be much different.

**Comment by Paakkit** [comment]
Uh, all the checks immediately failed.
I guess there's eventually some additional trick to be done and added to the PR?

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0-67-gc2de37f
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=rtklib-ci-builds&prefix=v1.0-67-gc2de37f/166-linux/)
+ s3://rtklib-ci-builds/v1.0-67-gc2de37f/166-linux

File: src/rcv/swiftnav.c [hunk 1]
```diff
@@ -650,9 +837,107 @@ static int decode_gpsnav(raw_t *raw) {
...
+    trace(2, "Unrecognised code %d for C%02d\n", eph.code, prn);
+    return -1;
+  }
+
+  decode_bdsnav_common(puiTmp - 2, &eph);
+
+  eph.ttr = raw->time;
+
+  if (!strstr(raw->opt, "EPHALL")) {
+    if ((eph.iode == raw->nav.eph[sat - 1].iode) &&
```
Comment (Paakkit) [inline src/rcv/swiftnav.c]: Any suggestions for this part?
We don't have proper usage of iode & iodc from FW.

Comment (Paakkit) [reply src/rcv/swiftnav.c]: For BDS that is.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0-67-gc2de37f
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=rtklib-ci-builds&prefix=v1.0-67-gc2de37f/166-osx/)
+ s3://rtklib-ci-builds/v1.0-67-gc2de37f/166-osx

File: src/rcv/swiftnav.c [hunk 2]
```diff
@@ -997,7 +1178,9 @@ extern int input_sbpjsonf(raw_t *raw, FILE *fp) {
   /* avoid parsing the payload if the message isn't supported in the first place
    */
   if ((uMsgType != ID_MEASEPOCH) && (uMsgType != ID_MSGEPHGPS_DEP1) &&
-      (uMsgType != ID_MSGEPHGPS) && (uMsgType != ID_MSGEPHGLO) &&
+      (uMsgType != ID_MSGEPHGPS_DEP2) && (uMsgType != ID_MSGEPHGPS) &&
+      (uMsgType != ID_MSGEPHBDS) && (uMsgType != ID_MSGEPHGAL) &&
+      (uMsgType != ID_MSGEPHGLO_DEP1) && && (uMsgType != ID_MSGEPHGLO) &&
```
[the remark this answers is not in the document]
Comment (Paakkit) [reply src/rcv/swiftnav.c]: Silly me,
Now it looks like it's passing the checks.

File: src/rcv/swiftnav.c (hunk 1 above)
Comment (anth-cole) [reply src/rcv/swiftnav.c]: I don't believe that BDS has the concept of IODE, the age of corrections has been used but is not equivalent.

**Comment by Paakkit** [comment]
@mbavaro 
I split this PR into 2 new PRs:
Step 1. https://github.com/swift-nav/RTKLIB/pull/28
Step 2. https://github.com/swift-nav/RTKLIB/pull/29

I am having difficult time finding recent Beidou rinex data to do a comparison.
Probably can just rely on what Piksi decodes and do comparison against that.

The Beidou IODE check still needs to be changed.

**Comment by mbavaro** [comment]
I am wondering if we could just generate a dummy IODE as `TOW_s / 3600` (increments by 1 every hour).. woulnd't protect against mixed subframe sets, but at least would have _some_ meaning

**Comment by Paakkit** [comment]
I put up a PR for iode: https://github.com/swift-nav/piksi_firmware_private/pull/2016
Now it increments every 30 minutes, allowing us to see out of schedule ephemeris updates between the regular full hours.

**Comment by Paakkit** [comment]
Closing.
This PR was split into 2 separate PRs:
https://github.com/swift-nav/RTKLIB/pull/28
https://github.com/swift-nav/RTKLIB/pull/29

**Pull request closed without merging**